src/main/java/com/tabnine/UninstallListener.kt

```diff
@@ -0,0 +1,62 @@
+package com.tabnine
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor
+import com.intellij.ide.plugins.PluginStateListener
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.util.text.SemVer
+import com.tabnine.binary.BinaryRequestFacade
+import com.tabnine.binary.requests.uninstall.UninstallRequest
+import com.tabnine.general.StaticConfig
+import com.tabnine.general.readTempTabninePluginZip
+import com.tabnine.lifecycle.UninstallReporter
+import java.nio.file.Paths
+
+private const val TABNINE_JAR_NAME = "TabNine-"
+private const val JAR_SUFFIX = ".jar"
+private val TABNINE_JAR_REGEX = """^$TABNINE_JAR_NAME\d+\.\d+\.\d+.jar""".toRegex()
+
+class UninstallListener(private val facade: BinaryRequestFacade, private val uninstallReporter: UninstallReporter) :
+    PluginStateListener {
+
+    override fun install(descriptor: IdeaPluginDescriptor) {
+        // nothing to do here
+    }
+
+    override fun uninstall(descriptor: IdeaPluginDescriptor) {
+        if (descriptor.pluginId?.let { it == StaticConfig.TABNINE_PLUGIN_ID } == false) {
```

Comment:
try catch on thie method, so that no "error" pops up to the user

Reply:
wdym, just wrap everything with try catch?

Reply:
yes